src/olympia/api/tests/test_authentication.py

```diff
@@ -0,0 +1,231 @@
+from calendar import timegm
+import mock
+import json
+
+from django.core.urlresolvers import reverse
+from django.test import RequestFactory
+
+import jwt
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from olympia.amo.helpers import absolutify
+from olympia.amo.tests import TestCase, WithDynamicEndpoints
+from olympia.api.authentication import (
+    JSONWebTokenAuthentication, JWTKeyAuthentication)
+from olympia.api.tests.test_jwt_auth import JWTAuthKeyTester
+from olympia.users.models import UserProfile
+
+
+class JWTKeyAuthTestView(APIView):
+    """
+    This is an example of a view that would be protected by
+    JWTKeyAuthentication, used in TestJWTKeyAuthProtectedView below.
+    """
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JWTKeyAuthentication]
+
+    def get(self, request):
+        return Response('some get response')
+
+    def post(self, request):
+        return Response({'user_pk': request.user.pk})
+
+
+class TestJWTKeyAuthentication(JWTAuthKeyTester):
+    fixtures = ['base/addon_3615']
+
+    def setUp(self):
+        super(TestJWTKeyAuthentication, self).setUp()
+        self.factory = RequestFactory()
+        self.auth = JWTKeyAuthentication()
+        self.user = UserProfile.objects.get(email='del@icio.us')
+
+    def request(self, token):
```

Comment:
I agree, but I thought it was more readable this way. I try to avoid adding conditions in tests helpers - sometimes it means the tests are not DRY, but in tests, I prefer being explicit, even if that means repeating myself.